### PR TITLE
AIX does not allow Process.getsid(pid) if pid is in a different session

### PR DESCRIPTION
**Pull request by odaira**
Please push this pull request back if it is too platform-specific.

### CODE REVIEW

File: core/process/setsid_spec.rb [hunk 1]
```diff
@@ -21,13 +21,17 @@
       write.close
       read2.close
       pgid_child = read.gets
+      pgid_child = Integer(pgid_child)
```
Comment (eregon) [inline core/process/setsid_spec.rb]: Can you merge this with the line above?

**Comment by eregon** [comment]
This looks fine, please address the little nitpick :smiley:

File: core/process/setsid_spec.rb (hunk 1 above)
Comment (odaira) [inline core/process/setsid_spec.rb]: Do you mean moving "pgid_child = Integer(pgid_child)" into the "platform_is_not :aix" block? But pgid_child is used not only at line 30 but also at line 35.

Comment (eregon) [inline core/process/setsid_spec.rb]: No, just merging with the line 23 above in a single expression:
```ruby
pgid_child = Integer(read.gets)
```

Comment (eregon) [inline core/process/setsid_spec.rb]: I'll do it :)

**Comment by eregon** [comment]
Thank you for your fix!

File: core/process/setsid_spec.rb (hunk 1 above)
Comment (odaira) [inline core/process/setsid_spec.rb]: Ah, I see. Thanks!

**Pull request merged**